aidial_assistant/chain/history.py

```diff
@@ -91,3 +123,63 @@ def to_best_effort_messages(
         )
 
         return messages
+
+    async def trim(
+        self, max_prompt_tokens: int, model_client: ModelClient
+    ) -> "History":
+        extra_results_callback = ModelExtraResultsCallback()
+        stream = model_client.agenerate(
+            self.to_protocol_messages(),
```

Comment:
This deserves a comment why you don't include assistant system message when you trim the history.

Reply:
Added comment.